### PR TITLE
DOC: Add module template

### PR DESCRIPTION
**Pull request by takanori-pskq**
Related to #13114. Added `:toctree:` to the lists of attributes, functions, classes, to generate the stub pages of all the members.

### CODE REVIEW

**Comment by mattip** [comment]
Could you point to some rendered documentation that uses the template?

**Comment by takanori-pskq** [comment]
`module.rst` template is used in `numpy.distutils.ccompiler`
* before: https://numpy.org/devdocs/reference/generated/numpy.distutils.ccompiler.html
* after: https://16607-908607-gh.circle-artifacts.com/0/doc/build/html/reference/generated/numpy.distutils.ccompiler.html

`numpy.distutils.ccompiler_opt`, `numpy.distutils.exec_command`, and `numpy.lib.format` also use this template.

File: doc/source/_templates/autosummary/module.rst [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{% extends "!autosummary/module.rst" %}
```
Comment (eric-wieser) [inline doc/source/_templates/autosummary/module.rst:1]:
```suggestion
{% extends "!autosummary/module.rst" %}

{# This file is almost the same as the default, but adds :toctree: to the autosummary directives.
   The original can be found at `sphinx/ext/autosummary/templates/autosummary/module.rst`. #}
```
or whatever the comment syntax is for templating

**Comment by charris** [comment]
close/reopen

**Comment by charris** [comment]
The failed test can be ignored. Ping @mattip.

**Comment by mattip** [comment]
Thanks @takanori-pskq

**Pull request merged**